event-session-signup/Code.gs

```diff
@@ -0,0 +1,176 @@
+/**
+ * A special function that inserts a custom menu when the spreadsheet opens.
+ */
+function onOpen() {
+  SpreadsheetApp.getUi().createMenu('Conference')
+    .addItem('Set up conference', 'setUpConference_')
+    .addToUi();
+}
+
+/**
+ * A set-up function that uses the conference data in the spreadsheet to create
+ * Google Calendar events, a Google Form, and a trigger that allows the script
+ * to react to form responses.
+ */
+function setUpConference_() {
+  if (ScriptProperties.getProperty('calId')) {
+    Browser.msgBox('Your conference is already set up. Look in Google Drive!');
+  }
+  var ss = SpreadsheetApp.getActive();
+  var sheet = ss.getSheetByName('Conference Setup');
+  var range = sheet.getDataRange();
+  var values = range.getValues();
+  setUpCalendar_(values, range);
+  setUpForm_(ss, values);
+  ScriptApp.newTrigger('onFormSubmit').forSpreadsheet(ss).onFormSubmit()
+      .create();
+  ss.removeMenu('Conference');
+}
+
+/**
+ * Creates a Google Calendar with events for each conference session in the
+ * spreadsheet, then writes the event IDs to the spreadsheet for future use.
+ * @param {Array<string[]>} values Cell values for the spreadsheet range.
+ * @param {Range} range A spreadsheet range that contains conference data.
+ */
+function setUpCalendar_(values, range) {
+  var cal = CalendarApp.createCalendar('Conference Calendar');
+  for (var i = 1; i < values.length; i++) {
```

Comment:
Might want a comment about why row 0 is skipped, or refactor calling code to shift the header row off so this code is less dependent on how the sheet is structured (better if values is *just* data here)